doc/import.rst

```diff
@@ -0,0 +1,81 @@
+Importing SpikeInterface
+========================
+
+SpikeInterface allows for the generation of powerful and reproducible spike sorting pipelines.
+Flexibility is built into the package starting from import to maximize the productivity of
+the developer and the scientist. Thus there are three ways that SpikeInterface and its components
+can be imported:
+
+
+Importing by Module
+-------------------
+
+Since each spike sorting pipeline involves a series of often repeated steps, many of the developers
+working on SpikeInterface recommend importing in a module by module fashion. This will allow you to
+keep track of your processing steps (preprocessing, postprocessing, quality metrics, etc.). This can
+be accomplished by:
+
+.. code-block:: python
+
+    import spikeinterface as si
+
+to import the :code:`core` module followed by:
+
+.. code-block:: python
+
+    import spikeinterface.extractors as se
+    import spikeinterface.preprocessing as spre
+    import spikeinterface.sorters as ss
+    import spikinterface.postprocessing as spost
+    import spikeinterface.qualitymetrics as sqm
+    import spikeinterface.exporters as sexp
+    import spikeinterface.comparsion as scmp
+    import spikeinterface.curation as scur
+    import spikeinterface.sortingcomponents as sc
+    import spikeinterface.widgets as sw
+
+to import any of the other modules you wish to use.
+
+The benefit of this approach is that it is lighter and faster than importing the whole package and allows
+you to choose which of the modules you actually want to use. If you don't plan to export the results out of
+SpikeInterface then you don't have to :code:`import spikeinterface.exporters`. Additionally the documentation
+of the package is set-up in a modular fashion, so if you have a problem with the module :code:`spikeinterface.curation`,
+you will know to go to the :code:`curation` section of this documention. The disadvantage of this approach is
+that you have more aliases to keep track of.
+
+
+Flat Import
+-----------
+
+A second option is to import the SpikeInterface package in :code:`full` mode. This would be similar to
+what is seen with packages like NumPy (:code:`np`) or Pandas (:code:`pd`). To accomplish this one does:
+
+
+.. code-block:: python
+
+    import spikeinterface.full as si
+
+
+This import statement will import all of the SpikeInterface modules as one flattened module.
+Note that importing :code:`spikeinterface.full` will take a few extra seconds, because some modules use
```

Comment:
I also think this is not true. The compilation is performed at import time when you have the package. Because of the try-except block.

Reply:
I copied that from your guys old docs. :)

Reply:
I saw that when I was at the end of the PR. Feel free to ignore this and I will correct it later.

Reply:
Or you re-write all of spikeinterface so it this one line of documentation is true :P. Pick your path wisely.

Reply:
Long term goal.
I moved try-excepts here and there when I find them.

The biggest thing is the numba import with try-except I believe.

Reply:
I don't have the docs on my hand, but I read somewhere that try-except is super fast as long as you never trigger the except. Once you trigger the except it becomes terribly slow. But I'll need to read up on it more.

Reply:
It is not the try-except structure. Is the fact that if you have the package in the try part then the package is imported.
We want to delay imports until they are actually used and not load them at import time (googel lazy loading).

For most packages we just put the import in the right place. But for plotting libraries and numba that is difficult. I don't remember what was the deal in the extractor module.

Reply:
I see what you mean for this.

Reply:
I also think that this statement on numba on numba is not true. I do not think numba is making compiling but numba is doing something. Maybe we should remove this numba related story and focus on the main message : "everythink is flat and and bit slower"

Reply:
I think we should remove the numba story yes